lib/px/block/pxblock.lua

```diff
@@ -57,62 +70,85 @@ function M.load(config_file)
 
         px_client.send_to_perimeterx('block', details);
         if px_config.block_enabled then
-            ngx.header["Content-Type"] = 'text/html';
-            if px_config.custom_block_url then
-                if not px_config.redirect_on_custom_url then
-                    local res = ngx.location.capture(px_config.custom_block_url)
-                    if res.truncated or res.status >= 300 then
-                        ngx.status = 500
-                        ngx_say('Unable to fetch custom block url. Status: ' .. tostring(res.status))
-                        ngx_exit(ngx.OK)
-                    end
-                    local body = ''
-                    if ngx.ctx.px_action == 'j' then
-                        body = ngx.ctx.px_action_data
-                    else
-                        body = res.body
-                        if ngx.ctx.px_action == 'c' then
-                            body = string_gsub(res.body, '</head>', inject_captcha_script(vid, uuid) .. '</head>', 1);
-                            body = string_gsub(body, '::BLOCK_REF::', uuid);
-                        end
-                    end
-                    ngx.status = ngx_HTTP_FORBIDDEN;
-                    ngx_say(body);
-                    ngx_exit(ngx.OK);
+            if ngx.ctx.px_cookie_origin == "header" then
+                local html = px_template.get_template("captcha.mobile", details.block_uuid, vid)
+                local collectorUrlBase = ''
+                if px_config.ssl_enabled then
+                    collectorUrlBase = 'https://sapi-'
                 else
-                    -- handling custom block url: create redirect url with original request url, vid and uuid as query params to use with captcha_api
-                    local req_query_param = ngx.req.get_uri_args()
-                    local enc_url, enc_args
-                    local original_req_url = ngx.var.uri
-                    if req_query_param then
-                        enc_args = ngx_encode_args(req_query_param)
-                        enc_url = ngx_endcode_64(original_req_url .. '?' .. enc_args)
-                    end
-                    local redirect_url = px_config.custom_block_url .. '?url=' .. enc_url .. '&uuid=' .. uuid .. '&vid=' .. vid
-                    px_logger.debug('Redirecting to custom block page: ' .. redirect_url)
-                    ngx_redirect(redirect_url, ngx_HTTP_TEMPORARY_REDIRECT)
+                    collectorUrlBase = 'http://sapi-'
                 end
-            else
+                local collectorUrl = collectorUrlBase .. string.lower(px_config.px_appId) .. '.perimeterx.net'
```

Comment:
the config should contain certain derivative, baseUrl which its default value is sapi-<app_id>.perimeterx.net and can be override 
Here it should only be set to https/http (that more or like what I think)